src/python/zquantum/vqe/singlet_uccsd.py

```diff
@@ -0,0 +1,131 @@
+from zquantum.core.interfaces.ansatz import Ansatz
+from zquantum.core.interfaces.ansatz_utils import (
+    ansatz_property,
+    invalidates_parametrized_circuit,
+)
+from zquantum.core.circuit import Circuit
+from openfermion.utils import uccsd_singlet_paramsize, uccsd_singlet_generator
+from overrides import overrides
+from .utils import exponentiate_fermion_operator, create_layer_of_gates
+from pyquil import Program
+from pyquil.gates import X
+from typing import List, Optional
+import sympy
+import numpy as np
+
+
+class SingletUCCSDAnsatz(Ansatz):
+
+    supports_parametrized_circuits = False
+    transformation = ansatz_property("transformation")
+
+    def __init__(
+        self,
+        n_spatial_orbitals: int,
+        n_alpha: int,
+        n_layers: int,
+        transformation: str = "Jordan-Wigner",
+    ):
```

Comment:
@mstechly I think `n_layers` should always be the first import for consistency, but feel free to disagree.

Reply:
Adressed in [75170d8](https://github.com/zapatacomputing/z-quantum-vqe/pull/5/commits/75170d8c5c38ec0adc9f66321ca231a9fec080cf).